### PR TITLE
feat(payment): PAYPAL-2910 Move googlepay-braintree from core to google-pay-integration package

### PR DESCRIPTION
**Pull request by bc-nick**
## What?

Move googlepay-braintree from core to `google-pay-integration` package

## Why?

To separate the logic of integrations

## Testing / Proof

With threeDSecure

https://github.com/bigcommerce/checkout-sdk-js/assets/99336044/315c7926-8c6c-4674-b42e-8894bf70b924

Without threeDSecure

https://github.com/bigcommerce/checkout-sdk-js/assets/99336044/bdbc3d5b-c9c0-440d-b85f-6ac982555566

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

**Comment by bc-peng** [comment]
Could you please create e2e tests to cover the package like the existing Google Pay (authroize.net) e2e tests?

**Comment by bc-nick** [comment]
> Could you please create e2e tests to cover the package like the existing Google Pay (authroize.net) e2e tests?

@bc-peng You mean creating an e2e test in `checkout-js` to cover this Google Pay Braintree payment strategy. Am I right?

**Comment by bc-peng** [comment]
> > Could you please create e2e tests to cover the package like the existing Google Pay (authroize.net) e2e tests?
> 
> 
> 
> @bc-peng You mean creating an e2e test in `checkout-js` to cover this Google Pay Braintree payment strategy. Am I right?

Yes, you are right.
You know how easy it is to break a payment method nowadays.

**Comment by bc-nick** [comment]
@bc-peng e2e test for covering GP Braintree: https://github.com/bigcommerce/checkout-js/pull/1516

File: packages/braintree-utils/src/index.ts [hunk 1]
```diff
@@ -41,6 +45,10 @@ export {
     getBraintreeLocalPaymentMock,
     getBraintree,
     getDeviceDataMock,
+				getModuleCreatorMock,
```
Comment (andriiVitvitskyi1990) [inline packages/braintree-utils/src/index.ts:48]: Please fix indentation

**Review by bc-dronov — APPROVED** [review]
Looks good for me.

File: packages/braintree-utils/src/index.ts (hunk 1 above)
Comment (bc-nick) [reply packages/braintree-utils/src/index.ts]: done

**Review by andriiVitvitskyi1990 — APPROVED** [review]

File: packages/braintree-utils/src/braintree.ts [hunk 2]
```diff
@@ -386,10 +387,11 @@ export interface BraintreeThreeDSecureCreatorConfig extends BraintreeModuleCreat
 export interface BraintreeThreeDSecureOptions {
     nonce: string;
     amount: number;
-    challengeRequested: boolean;
+    challengeRequested?: boolean;
```
Comment (vitalii-koshovyi) [inline packages/braintree-utils/src/braintree.ts:390]: Why does it become an optional field?

**Review by vitalii-koshovyi — APPROVED** [review]

**Review by bc-peng — APPROVED** [review]
LGTM. 👏👏

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**